src/Npgsql/ConnectorPool.cs

```diff
@@ -329,11 +336,11 @@ await Task.WhenAny(tcs.Task, Task.Delay(timeLeft, delayCancellationToken.Token))
                                 if (timeout.IsSet)
                                 {
                                     var timeLeft = timeout.TimeLeft;
-                                    if (timeLeft <= TimeSpan.Zero || !tcs.Task.Wait(timeLeft))
+                                    if (timeLeft <= TimeSpan.Zero || !tcs.Task.Wait((int)timeLeft.TotalMilliseconds, cancellationToken))
```

Comment:
What is a reason to get milliseconds and convert them to `int`? If the value is larger than `int.MaxValue` there could be no exception at all because it could be trimmed.

Reply:
This is the maximum expected in Task.Wait, it would be good to indeed add a argument exception like the overload taking just a TimeSpan does, will do.
```
public bool Wait(TimeSpan timeout)
{
      long totalMilliseconds = (long) timeout.TotalMilliseconds;
      if (totalMilliseconds < -1L || totalMilliseconds > (long) int.MaxValue)
        ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.timeout);
      return this.Wait((int) totalMilliseconds, new CancellationToken());
}
```
There's just no overload taking a TimeSpan and a CancellationToken

Reply:
To add this could have already happened before with an insanely high NpgsqlTimeout, it's just made visible now.

Reply:
> There's just no overload taking a TimeSpan and a CancellationToken

Then it's a good idea to open a pull request in Core FX repo and add it to .NET Standard 2.1 too.

Reply:
Done